src/nvim/api/vim.c

```diff
@@ -2451,3 +2451,71 @@ void nvim_del_user_command(String name, Error *err)
 {
   nvim_buf_del_user_command(-1, name, err);
 }
+
+/// Parse command line.
+///
+/// Doesn't check the validity of command arguments.
+///
+/// @param str       Command line string to parse. Cannot contain "\n".
+/// @param opts      Optional parameters. Reserved for future use.
+/// @param[out] err  Error details, if any.
+/// @return Dictionary containing command information, with these keys:
+///         - cmd: (string) command name.
+///         - line1: (number) starting line of command range.
+///         - line2: (number) final line of command range.
+///         - force: (boolean) whether command contains a bang.
+///         - args: (array) command arguments.
+Dictionary nvim_parse_cmdline(String str, Dictionary opts, Error *err)
```

Comment:
* Let's introduce "cmd" as an API noun and stick with it. This opens the door for `nvim_exec_cmd({list})`, then we can deprecate `nvim_command()` and maybe even `nvim_exec()`.
* "cmdline" implies the UI for entering cmds, which isn't really the case here.

```suggestion
Dictionary nvim_parse_cmd(String str, Dictionary opts, Error *err)
```